src/rdkafka_transport.c

```diff
@@ -762,6 +762,24 @@ int rd_kafka_transport_ssl_ctx_init (rd_kafka_t *rk,
                 }
 	}
 
+	/*
+	 * If set, can override the above, as well as do other configurations
+	 * with the context object (none of the above is necessary to use this)
+	 */
+	if (rk->rk_conf.ssl_ctx_cb) {
+		rd_kafka_dbg(rk, SECURITY, "SSL",
+		            "Using user-defined SSL_CTX callback");
+
+		r = rk->rk_conf.ssl_ctx_cb(rk,
+		                           (void *)ctx,
+		                           rk->rk_conf.opaque);
+
+		if (r < 0) {
```

Comment:
the return value type is rd_kafka_resp_err_t  (use properly typed variable) and should be checked for ` != RD_KAFKA_RESP_ERR_NO_ERROR`.
Also, in this case, print the error code in the errstr and indicate where the error came from:
```
rd_snprintf(errstr, errstr_size, "User-defined SSL_CTX callback failed: %s", rd_kafka_err2str(err));
```